library/camellia.c

```diff
@@ -495,6 +510,11 @@ int mbedtls_camellia_crypt_ecb( mbedtls_camellia_context *ctx,
 {
     int NR;
     uint32_t *RK, X[4];
+    CAMELLIA_VALIDATE_RET( ctx != NULL );
+    CAMELLIA_VALIDATE_RET( mode == MBEDTLS_CAMELLIA_ENCRYPT ||
```

Comment:
I have reservations about this. On the one hand it is correct, but on the other hand it may be better to state in the documentation, that this parameter is ignored, and not validate it at all.

Reply:
Sorry @k-stachowiak, I didn't get your point here - mode must be one of `MBEDTLS_CAMELLIA_ENCRYPT` or `MBEDTLS_CAMELLIA_DECRYPT`. This is a precondition stated in the documentation, and accordingly checked here in case parameter validation is enabled.

Reply:
We have `( (void) mode );` few lines below though. Maybe this is not supposed to be visible to the client; I think this depends on some general documentation strategy, but technically, these checks aren't necessary, and I was wondering if we should treat this case differently for that reason.